hydra/redisgraph_demo1.py

```diff
@@ -0,0 +1,258 @@
+"""
+*Readme for this file*:
+ It contains a graph which contain the api structure of api doc
+ It design with the help of endpoints and its operation
+ Design:
+  url or id => endpoints[] => collection(get) and classes(put or post)
+                             ((if collection then => classes=> members(get))
+                             elif classes then =>members(get))
+
+
+"""
+
+
+import redis
+from redisgraph import Graph, Node, Edge
+import urllib
+import json
+import re
+from re import compile as regex
+from httplib2 import Http
+
+redis_con = redis.Redis(host='localhost', port=6379)
+redis_graph = Graph("apidoc", redis_con)
+"""
+getting the apidoc
+"""
+
+
+class _MemCache(dict):
+
+    def __nonzero__(self):
+        # even if empty, a _MemCache is True
+        return True
+
+    def set(self, key, value):
+        self[key] = value
+
+    def delete(self, key):
+        if key in self:
+            del self[key]
+
+
+DEFAULT_HTTP_CLIENT = Http(_MemCache())
+http = DEFAULT_HTTP_CLIENT
+APIDOC_RE = regex(
+    r'^<([^>]*)>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"$')
+
+
+def final_file(url):
+    response = urllib.urlopen(url)
```

Comment:
Please provide proper Docstrings for each function. Read this for example http://epydoc.sourceforge.net/docstrings.html